crawl-ref/source/dat/des/branches/lair.des

```diff
@@ -440,6 +440,63 @@ MAP
     ..@...
 ENDMAP
 
+NAME:    mainiacjoe_lair_zoo
+TAGS:    lair_entry transparent
+TAGS:    no_monster_gen no_item_gen
+ORIENT:  float
+SHUFFLE: NALPS
+NSUBST:  P = 1:P / *:`
+SUBST:   P = Pp
+KMONS:   P = anaconda
+KMONS:   p = shock serpent
```

Comment:
The second subst on these isn't necessary, you can just do KMONS: P = anaconda / shock serpent. Likewise, if you're going to always make three cages use floor_dirt you can just make L,  P, and N all nsubst to ' and define the ftile at the end. No need to do it in three places.